mwclient/client.py

```diff
@@ -941,10 +1028,15 @@ def upload(self, file=None, filename=None, description='', ignore=False,
             if not hasattr(file, 'read'):
                 file = open(file, 'rb')
 
+            # Narrowing the type of file from Union[str, io.BufferedReader] to just
```

Comment:
it was never `Union[str, io.BufferedReader]`, though?

Reply:
Fixed in ab9d64c